modular_skyrat/modules/modular_implants/code/soulcatcher/soulcatcher_component.dm

```diff
@@ -0,0 +1,273 @@
+ * * message_sender - The person that is sending the message. This is not required.
+ * * emote - Is the message sent an emote or not?
+ */
+
+/datum/soulcatcher_room/proc/send_message(message_to_send, message_sender, emote = FALSE)
+	if(!message_to_send) //Why say nothing?
+		return FALSE
+
+	var/sender_name = ""
+	if(message_sender)
+		sender_name = "[message_sender] "
+
+	var/datum/asset/spritesheet/sheet = get_asset_datum(/datum/asset/spritesheet/chat)
+	var/tag = sheet.icon_tag("nif-soulcatcher")
+	var/soulcatcher_icon = ""
+
+	if(tag)
+		soulcatcher_icon = tag
+
+	var/message = ""
+	if(!emote)
+		message = "<font color=[SOULCATCHER_SAY_COLOR]>\ [soulcatcher_icon] <b>[sender_name]</b>says, \"[message_to_send]\"</font>"
+		log_say("[sender_name] in [name] soulcatcher room said: [message_to_send]")
+	else
+		message = "<font color=[SOULCATCHER_EMOTE_COLOR]>\ [soulcatcher_icon] <b>[sender_name]</b>[message_to_send]</font>"
+		log_emote("[sender_name] in [name] soulcatcher room emoted: [message_to_send]")
+
+	for(var/mob/living/soulcatcher_soul/soul in current_souls)
+		if((emote && !soul.internal_sight) || (!emote && !soul.internal_hearing))
+			continue
+
+		to_chat(soul, message)
+
+	relay_message_to_soulcatcher(message)
+	return TRUE
+
+/// Relays a message sent from the send_message proc to the parent soulcatcher datum
+/datum/soulcatcher_room/proc/relay_message_to_soulcatcher(message)
+	if(!message)
+		return FALSE
+
+	var/datum/component/soulcatcher/recepient_soulcatcher = master_soulcatcher.resolve()
+	recepient_soulcatcher.recieve_message(message)
+	return TRUE
+
+/datum/soulcatcher_room/Destroy(force, ...)
+	for(var/mob/living/soulcatcher_soul in current_souls)
+		qdel(soulcatcher_soul)
+		current_souls -= soulcatcher_soul
+
+	return ..()
+
+/mob/dead/observer/verb/join_soulcatcher()
+	set name = "Enter Soulcatcher"
+	set category = "Ghost"
+
+	var/list/joinable_soulcatchers = GLOB.soulcatchers.Copy()
+	for(var/datum/component/soulcatcher/soulcatcher in joinable_soulcatchers)
+		if(soulcatcher.ghost_joinable)
+			continue
+		joinable_soulcatchers -= (soulcatcher)
+
+	if(!length(joinable_soulcatchers))
+		to_chat(src, span_warning("No soulcatchers are joinable."))
+		return FALSE
+
+	var/datum/component/soulcatcher/soulcatcher_to_join = tgui_input_list(src, "Chose a soulcatcher to join", "Enter a soulcatcher", joinable_soulcatchers)
+	if(!soulcatcher_to_join || !(soulcatcher_to_join in joinable_soulcatchers))
+		return FALSE
+
+	var/list/rooms_to_join = soulcatcher_to_join.soulcatcher_rooms.Copy()
+	for(var/datum/soulcatcher_room/room in rooms_to_join)
+		if(room.joinable)
+			continue
+		rooms_to_join -= room
+
+	var/datum/soulcatcher_room/room_to_join
+	if(length(rooms_to_join) < 1)
+		return FALSE
+
+	if(length(rooms_to_join) == 1)
+		room_to_join = rooms_to_join[1]
+
+	else
+		room_to_join = tgui_input_list(src, "Chose a room to enter", "Enter a room", rooms_to_join)
+
+	if(!room_to_join)
+		return FALSE
```

Comment:
I feel like a bit of feedback with a `to_chat()` could potentially be pertinent here? Not sure,